### ISSUE TITLE
azure-webapp-maven-plugin:2.9.0 seems to have errors

### ISSUE THREAD

**Issue by n-y-kim** [Microsoft/azure-maven-plugins#2287]
### Plugin name and version
azure-webapp-maven-plugin:2.9.0

### Plugin configuration in your `pom.xml`
```xml
		        <groupId>com.microsoft.azure</groupId>
		        <artifactId>azure-webapp-maven-plugin</artifactId>
		        <version>2.9.0</version>
		        <configuration>
		            <schemaVersion>v2</schemaVersion>
		            <subscriptionId>{MYSUBSCRIPTIONID}</subscriptionId>
		            <resourceGroup>{MYRESOURCEGROUP}</resourceGroup>
		            <appName>{MYAPPNAME}</appName>
		            <pricingTier>S1</pricingTier>
		            <region>koreacentral</region>
		            <appServicePlanName>{MYAPPSERVICEPLAN}</appServicePlanName>
		            <appServicePlanResourceGroup>{MYRESOURCEGROUP}</appServicePlanResourceGroup>
		            <runtime>
		                <os>Linux</os>
		                <javaVersion>Java 17</javaVersion>
		                <webContainer>Java SE</webContainer>
		            </runtime>
		            <deployment>
		                <resources>
		                    <resource>
		                        <directory>${project.basedir}/target</directory>
		                        <includes>
		                            <include>*.jar</include>
		                        </includes>
		                    </resource>
		                </resources>
		            </deployment>
		        </configuration>
```

### Expected behavior
```
mvn package azure-webapp:deploy
```

The command successfully deploying web app.

### Actual behavior
Showing the error like below:

```
[WARNING] 'Content-Type' not found. Returning default encoding: JSON
[ERROR] Status code 401, (empty body)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  23.930 s
[INFO] Finished at: 2023-04-29T21:38:02+09:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.microsoft.azure:azure-webapp-maven-plugin:2.9.0:deploy (default-cli) on project api: deploy to Azure Web App(appsvc-roadshow-test-api) with resource creation or updating: Status code 401, (empty body) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.microsoft.azure:azure-webapp-maven-plugin:2.9.0:deploy (default-cli) on project api: deploy to Azure Web App(appsvc-roadshow-test-api) with resource creation or updating
```

I tried to get detail log with mvn -e key, and it did not show any meaningful message either.

<details>
<summary>detail error message:</summary>

```
 at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:347)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:330)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:175)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:76)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:163)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:160)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:827)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:272)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:195)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: deploy to Azure Web App(appsvc-roadshow-test-api) with resource creation or updating
    at com.microsoft.azure.maven.AbstractAzureMojo.onMojoError (AbstractAzureMojo.java:545)
    at com.microsoft.azure.maven.AbstractAzureMojo.execute (AbstractAzureMojo.java:311)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:342)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:330)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:175)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:76)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:163)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:160)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:827)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:272)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:195)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: com.microsoft.azure.toolkit.lib.common.operation.OperationException: deploy to Azure Web App(appsvc-roadshow-test-api) with resource creation or updating
    at com.microsoft.azure.toolkit.lib.common.operation.AzureOperationAspect.afterThrowing (AzureOperationAspect.java:94)
    at com.microsoft.azure.toolkit.lib.common.operation.AzureOperationAspect.afterThrowing (AzureOperationAspect.java:49)
    at com.microsoft.azure.toolkit.lib.appservice.task.DeployWebAppTask.doExecute (DeployWebAppTask.java:84)
    at com.microsoft.azure.maven.webapp.DeployMojo.deploy (DeployMojo.java:141)
    at com.microsoft.azure.maven.webapp.DeployMojo.doExecute (DeployMojo.java:87)
    at com.microsoft.azure.maven.AbstractAzureMojo.execute (AbstractAzureMojo.java:307)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:342)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:330)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:175)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:76)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:163)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:160)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:827)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:272)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:195)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: com.azure.core.exception.HttpResponseException: Status code 401, (empty body)
    at com.azure.core.implementation.http.rest.RestProxyBase.instantiateUnexpectedException (RestProxyBase.java:359)
    at com.azure.core.implementation.http.rest.AsyncRestProxy.lambda$ensureExpectedStatus$1 (AsyncRestProxy.java:115)
    at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext (FluxMapFuseable.java:113)
    at reactor.core.publisher.Operators$ScalarSubscription.request (Operators.java:2363)
    at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.request (FluxMapFuseable.java:171)
    at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.set (Operators.java:2159)
    at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onSubscribe (Operators.java:2033)
    at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onSubscribe (FluxMapFuseable.java:96)
    at reactor.core.publisher.MonoJust.subscribe (MonoJust.java:55)
    at reactor.core.publisher.InternalMonoOperator.subscribe (InternalMonoOperator.java:64)
    at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext (MonoFlatMap.java:157)
    at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext (FluxMapFuseable.java:129)
    at reactor.core.publisher.FluxHide$SuppressFuseableSubscriber.onNext (FluxHide.java:137)
    at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext (FluxMapFuseable.java:129)
    at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext (FluxMapFuseable.java:129)
    at reactor.core.publisher.FluxHide$SuppressFuseableSubscriber.onNext (FluxHide.java:137)
    at reactor.core.publisher.FluxMap$MapSubscriber.onNext (FluxMap.java:122)
    at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext (FluxOnErrorResume.java:79)
    at reactor.core.publisher.Operators$MonoSubscriber.complete (Operators.java:1802)
    at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext (MonoFlatMap.java:151)
    at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext (FluxMapFuseable.java:129)
    at reactor.core.publisher.Operators$MonoSubscriber.complete (Operators.java:1802)
    at reactor.core.publisher.MonoFlatMap$FlatMapInner.onNext (MonoFlatMap.java:249)
    at reactor.core.publisher.SerializedSubscriber.onNext (SerializedSubscriber.java:99)
    at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.onNext (FluxRetryWhen.java:174)
    at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext (FluxOnErrorResume.java:79)
    at reactor.core.publisher.Operators$MonoInnerProducerBase.complete (Operators.java:2629)
    at reactor.core.publisher.MonoSingle$SingleSubscriber.onComplete (MonoSingle.java:180)
    at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onComplete (MonoFlatMapMany.java:260)
    at reactor.core.publisher.FluxMap$MapSubscriber.onComplete (FluxMap.java:144)
    at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onComplete (FluxSwitchIfEmpty.java:85)
    at reactor.core.publisher.Operators$ScalarSubscription.request (Operators.java:2365)
    at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.set (Operators.java:2159)
    at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onSubscribe (Operators.java:2033)
    at reactor.core.publisher.MonoJust.subscribe (MonoJust.java:55)
    at reactor.core.publisher.Mono.subscribe (Mono.java:4490)
    at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onComplete (FluxSwitchIfEmpty.java:82)
    at reactor.core.publisher.FluxDoFinally$DoFinallySubscriber.onComplete (FluxDoFinally.java:128)
    at reactor.core.publisher.FluxHandle$HandleSubscriber.onComplete (FluxHandle.java:220)
    at reactor.core.publisher.FluxMap$MapConditionalSubscriber.onComplete (FluxMap.java:275)
    at reactor.core.publisher.FluxDoFinally$DoFinallySubscriber.onComplete (FluxDoFinally.java:128)
    at reactor.core.publisher.FluxHandleFuseable$HandleFuseableSubscriber.onNext (FluxHandleFuseable.java:208)
    at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext (FluxContextWrite.java:107)
    at reactor.core.publisher.Operators$MonoSubscriber.complete (Operators.java:1802)
    at reactor.core.publisher.MonoCollectList$MonoCollectListSubscriber.onComplete (MonoCollectList.java:129)
    at reactor.core.publisher.FluxPeek$PeekSubscriber.onComplete (FluxPeek.java:260)
    at reactor.core.publisher.FluxMap$MapSubscriber.onComplete (FluxMap.java:144)
    at reactor.netty.channel.FluxReceive.onInboundComplete (FluxReceive.java:413)
    at reactor.netty.channel.ChannelOperations.onInboundComplete (ChannelOperations.java:419)
    at reactor.netty.channel.ChannelOperations.terminate (ChannelOperations.java:473)
    at reactor.netty.http.client.HttpClientOperations.onInboundNext (HttpClientOperations.java:702)
    at reactor.netty.channel.ChannelOperationsHandler.channelRead (ChannelOperationsHandler.java:113)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:379)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:365)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead (AbstractChannelHandlerContext.java:357)
    at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead (CombinedChannelDuplexHandler.java:436)
    at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead (ByteToMessageDecoder.java:336)
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead (ByteToMessageDecoder.java:308)
    at io.netty.channel.CombinedChannelDuplexHandler.channelRead (CombinedChannelDuplexHandler.java:251)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:379)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:365)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead (AbstractChannelHandlerContext.java:357)
    at io.netty.handler.ssl.SslHandler.unwrap (SslHandler.java:1373)
    at io.netty.handler.ssl.SslHandler.decodeNonJdkCompatible (SslHandler.java:1247)
    at io.netty.handler.ssl.SslHandler.decode (SslHandler.java:1287)
    at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection (ByteToMessageDecoder.java:519)
    at io.netty.handler.codec.ByteToMessageDecoder.callDecode (ByteToMessageDecoder.java:458)
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead (ByteToMessageDecoder.java:280)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:379)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:365)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead (AbstractChannelHandlerContext.java:357)
    at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead (DefaultChannelPipeline.java:1410)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:379)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:365)
    at io.netty.channel.DefaultChannelPipeline.fireChannelRead (DefaultChannelPipeline.java:919)
    at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady (AbstractEpollStreamChannel.java:800)
    at io.netty.channel.epoll.EpollEventLoop.processReady (EpollEventLoop.java:499)
    at io.netty.channel.epoll.EpollEventLoop.run (EpollEventLoop.java:397)
    at io.netty.util.concurrent.SingleThreadEventExecutor$4.run (SingleThreadEventExecutor.java:997)

```
</details>

So I was searching for updates or issues and found this repo. Found that the latest version is 2.11.0, deleted the plugin at `pom.xml` and retried the steps with **`2.11.0`** version. Successfully deployed.

### Steps to reproduce the problem
I followed the Quick Start Quide for Java. The doc is using `2.9.0` ver, and if this version has error like I had I think it should be updated to a newer version. Please check if there are any issues in` 2.9.0` version.

**Comment by robertmcnees** [comment]
Can this issue can be closed now that PR #2486 has been merged?

**Comment by n-y-kim** [comment]
Yes @robertmcnees, thanks for the PR.
